core/src/storage_stage.rs

```diff
@@ -400,14 +398,13 @@ impl StorageStage {
                                     if statew.replicator_map.len() <= max_segment_index {
                                         statew
                                             .replicator_map
-                                            .resize(max_segment_index, HashSet::new());
+                                            .resize(max_segment_index, HashMap::new());
                                     }
                                     let proof_segment_index =
                                         (proof_entry_height / ENTRIES_PER_SEGMENT) as usize;
-                                    if proof_segment_index < statew.replicator_map.len() {
-                                        statew.replicator_map[proof_segment_index]
-                                            .insert(tx.account_keys[0]);
-                                    }
+                                    statew.replicator_map[proof_segment_index]
```

Comment:
Safe to remove this check? I guess counting on the fact that it's in the instruction stream here then it should be ok.